### PR TITLE
CM-963 comet java SDK - allow to pass epochs

### PR DESCRIPTION
**Pull request by LifeXplorer**

### CODE REVIEW

File: comet-java-client/src/main/java/ml/comet/experiment/Experiment.java [hunk 1]
```diff
@@ -114,8 +114,9 @@
      * @param metricName The name for the metric to be logged
      * @param metricValue The new value for the metric.  If the values for a metric are plottable we will plot them
      * @param step The current step for this metric, this will set the given step for this experiment
+     * @param epoch The current epoch for this metric, this will set the given epoch for this experiment
      */
-    void logMetric(String metricName, Object metricValue, long step);
+    void logMetric(String metricName, Object metricValue, long step, long epoch);
```
Comment (Nimrod007) [inline comet-java-client/src/main/java/ml/comet/experiment/Experiment.java:119]: please make step and epoch as optional (Long) 
add method overload to allow logging with and without steps, with and without epoch

**Review by Nimrod007 — CHANGES_REQUESTED** [review]

File: comet-java-client/src/main/java/ml/comet/experiment/Experiment.java (hunk 1 above)
Comment (LifeXplorer) [reply comet-java-client/src/main/java/ml/comet/experiment/Experiment.java:119]: @Nimrod007, at present there is 
```
    public void logMetric(String metricName, Object metricValue) 
```
that uses current step and epoch. And there are setters for step and epoch. Should I add separate methods like this?
```
    public void logMetricWithStep(String metricName, Object metricValue, long step) 
    public void logMetricWithEpoch(String metricName, Object metricValue, long epoch) 
```

Comment (Nimrod007) [reply comet-java-client/src/main/java/ml/comet/experiment/Experiment.java:119]: we want:
```
logMetric(String metricName, Object metricValue, Long step, Long epoch)
logMetric(String metricName, Object metricValue, Long step)
logMetric(String metricName, Object metricValue, Long epoch)
logMetric(String metricName, Object metricValue)
```
and overload 1 impl 
```
logMetric(String metricName, Object metricValue, Long step, Long epoch)
```

Comment (LifeXplorer) [reply comet-java-client/src/main/java/ml/comet/experiment/Experiment.java:119]: Updated as following:
```
logMetric(String metricName, Object metricValue, Long step, Long epoch)
logMetric(String metricName, Object metricValue, Long step)
logMetric(String metricName, Object metricValue)
```

File: comet-java-client/src/main/java/ml/comet/experiment/OnlineExperimentImpl.java [hunk 2]
```diff
@@ -261,15 +277,21 @@ public String getExperimentKey() {
         return Optional.ofNullable(this.experimentLink);
     }
 
+    @Override
+    public void logMetric(String metricName, Object metricValue, Long step) {
```
Comment (Nimrod007) [inline comet-java-client/src/main/java/ml/comet/experiment/OnlineExperimentImpl.java:281]: change 
```
Long step
```
to 
```
long step
```
or change all other epoch and step method parameters to Long

**Review by Nimrod007 — CHANGES_REQUESTED** [review]

File: comet-java-client/src/main/java/ml/comet/experiment/OnlineExperimentImpl.java (hunk 2 above)
Comment (LifeXplorer) [reply comet-java-client/src/main/java/ml/comet/experiment/OnlineExperimentImpl.java]: Nice catch! Updated to primitive.

**Review by Nimrod007 — APPROVED** [review]

**Pull request merged**